### PR TITLE
feat: add Casdoor single sign-on to alist

### PR DESCRIPTION
**Pull request by longxu0509**
add Casdoor single sign-on to alist

### CODE REVIEW

**Comment by anwen-anyi** [comment]
:joy: :joy: :joy: 这个 [casdoor.org](casdoor.org) 是付费的耶？刚准备测试一下试试看发现好像是付费 最低的 `US$9`

**Comment by nomeguy** [comment]
@anwen-anyi No, Casdoor is fully open-sourced at: https://github.com/casdoor/casdoor (also developed in Go) under Apache 2.0 and can be self-hosted freely. You can choose the paid hosting version if you don't want self-host

**Comment by anwen-anyi** [comment]
> @anwen-anyi No, Casdoor is fully open-sourced at: https://github.com/casdoor/casdoor (also developed in Go) under Apache 2.0 and can be self-hosted freely. You can choose the paid hosting version if you don't want self-host

Thanks, got it. What I saw was built using their official services, which are paid for.

File: internal/bootstrap/data/setting.go [hunk 1]
```diff
@@ -152,9 +152,13 @@ func InitialSettings() []model.SettingItem {
 
 		// SSO settings
 		{Key: conf.SSOLoginEnabled, Value: "false", Type: conf.TypeBool, Group: model.SSO, Flag: model.PUBLIC},
-		{Key: conf.SSOLoginplatform, Type: conf.TypeSelect, Options: "Github,Microsoft,Google,Dingtalk", Group: model.SSO, Flag: model.PUBLIC},
+		{Key: conf.SSOLoginplatform, Type: conf.TypeSelect, Options: "Casdoor,Github,Microsoft,Google,Dingtalk", Group: model.SSO, Flag: model.PUBLIC},
 		{Key: conf.SSOClientId, Value: "", Type: conf.TypeString, Group: model.SSO, Flag: model.PRIVATE},
 		{Key: conf.SSOClientSecret, Value: "", Type: conf.TypeString, Group: model.SSO, Flag: model.PRIVATE},
```
Comment (Hanooona) [inline internal/bootstrap/data/setting.go:157]: ****

Comment (Hanooona) [inline internal/bootstrap/data/setting.go:157]: ****

Comment (Hanooona) [inline internal/bootstrap/data/setting.go:157]: ****

Comment (Hanooona) [reply internal/bootstrap/data/setting.go:157]: Hi

Comment (Hanooona) [reply internal/bootstrap/data/setting.go:157]: #

Comment (Hanooona) [reply internal/bootstrap/data/setting.go:157]: S

Comment (Hanooona) [reply internal/bootstrap/data/setting.go:157]: M

Comment (Hanooona) [reply internal/bootstrap/data/setting.go:157]: H

Comment (Hanooona) [reply internal/bootstrap/data/setting.go:157]: G

Comment (Hanooona) [reply internal/bootstrap/data/setting.go:157]: Y

Comment (Hanooona) [reply internal/bootstrap/data/setting.go:157]: T

Comment (Hanooona) [reply internal/bootstrap/data/setting.go:157]: E

Comment (Hanooona) [reply internal/bootstrap/data/setting.go:157]: F

Comment (Hanooona) [reply internal/bootstrap/data/setting.go:157]: ****

Comment (Hanooona) [reply internal/bootstrap/data/setting.go:157]: T

Comment (Hanooona) [reply internal/bootstrap/data/setting.go:157]: R

Comment (Hanooona) [reply internal/bootstrap/data/setting.go:157]: @@

Comment (Hanooona) [reply internal/bootstrap/data/setting.go:157]: -

Comment (Hanooona) [reply internal/bootstrap/data/setting.go:157]: ****

Comment (Hanooona) [reply internal/bootstrap/data/setting.go:157]: []()

Comment (Hanooona) [reply internal/bootstrap/data/setting.go:157]: @longxu0509

Comment (Hanooona) [reply internal/bootstrap/data/setting.go:157]: []()

Comment (Hanooona) [reply internal/bootstrap/data/setting.go:157]: ****

Comment (Hanooona) [reply internal/bootstrap/data/setting.go:157]: []()

Comment (Hanooona) [reply internal/bootstrap/data/setting.go:157]: __

Comment (Hanooona) [reply internal/bootstrap/data/setting.go:157]: R

Comment (Hanooona) [reply internal/bootstrap/data/setting.go:157]: T

Comment (Hanooona) [reply internal/bootstrap/data/setting.go:157]: ****

Comment (Hanooona) [reply internal/bootstrap/data/setting.go:157]: ##

Comment (Hanooona) [reply internal/bootstrap/data/setting.go:157]: @nomeguy

Comment (Hanooona) [reply internal/bootstrap/data/setting.go:157]: []()

Comment (Hanooona) [reply internal/bootstrap/data/setting.go:157]: ##

Comment (Hanooona) [reply internal/bootstrap/data/setting.go:157]: #****

Comment (Hanooona) [reply internal/bootstrap/data/setting.go:157]: #- [ ]

Comment (Hanooona) [reply internal/bootstrap/data/setting.go:157]: ****

Comment (Hanooona) [reply internal/bootstrap/data/setting.go:157]: ****

Comment (Hanooona) [reply internal/bootstrap/data/setting.go:157]: @longxu0509

Comment (Hanooona) [reply internal/bootstrap/data/setting.go:157]: ****

Comment (Hanooona) [reply internal/bootstrap/data/setting.go:157]: ##

Comment (Hanooona) [reply internal/bootstrap/data/setting.go:157]: _[]()_

Comment (Hanooona) [reply internal/bootstrap/data/setting.go:157]: @longxu0509

Comment (Hanooona) [reply internal/bootstrap/data/setting.go:157]: @longxu0509

Comment (Hanooona) [reply internal/bootstrap/data/setting.go:157]: @anwen-anyi

Comment (Hanooona) [reply internal/bootstrap/data/setting.go:157]: @nomeguy

Comment (Hanooona) [reply internal/bootstrap/data/setting.go:157]: @ovear

Comment (Hanooona) [reply internal/bootstrap/data/setting.go:157]: @longxu0509

Comment (Hanooona) [reply internal/bootstrap/data/setting.go:157]: @longxu0509

**Review by Hanooona — COMMENTED** [review]
****

File: internal/bootstrap/data/setting.go (hunk 1 above)
Comment (Hanooona) [reply internal/bootstrap/data/setting.go:157]: @longxu0509

Comment (Hanooona) [reply internal/bootstrap/data/setting.go:157]: ##

Comment (Hanooona) [reply internal/bootstrap/data/setting.go:157]: D****

Comment (Hanooona) [reply internal/bootstrap/data/setting.go:157]: ****

**Review by Hanooona — COMMENTED** [review]
##

File: internal/bootstrap/data/setting.go [hunk 2]
```diff
@@ -152,9 +152,13 @@ func InitialSettings() []model.SettingItem {
 
 		// SSO settings
 		{Key: conf.SSOLoginEnabled, Value: "false", Type: conf.TypeBool, Group: model.SSO, Flag: model.PUBLIC},
-		{Key: conf.SSOLoginplatform, Type: conf.TypeSelect, Options: "Github,Microsoft,Google,Dingtalk", Group: model.SSO, Flag: model.PUBLIC},
+		{Key: conf.SSOLoginplatform, Type: conf.TypeSelect, Options: "Casdoor,Github,Microsoft,Google,Dingtalk", Group: model.SSO, Flag: model.PUBLIC},
```
[the remark this answers is not in the document]
Comment (Hanooona) [reply internal/bootstrap/data/setting.go:155]: ##_

[the remark this answers is not in the document]
Comment (Hanooona) [reply internal/bootstrap/data/setting.go:155]: Gasdoor

[the remark this answers is not in the document]
Comment (Hanooona) [reply internal/bootstrap/data/setting.go:155]: Gasdoor

[the remark this answers is not in the document]
Comment (Hanooona) [reply internal/bootstrap/data/setting.go:155]: ##- [ ]

[the remark this answers is not in the document]
Comment (Hanooona) [reply internal/bootstrap/data/setting.go:155]: ##****

[the remark this answers is not in the document]
Comment (Hanooona) [reply internal/bootstrap/data/setting.go:155]: 8f1f011fef11d17dac090609978f0ee2df2b1ceb

[the remark this answers is not in the document]
Comment (Hanooona) [reply internal/bootstrap/data/setting.go:155]: 8f1f011fef11d17dac090609978f0ee2df2b1ceb

[the remark this answers is not in the document]
Comment (Hanooona) [reply internal/bootstrap/data/setting.go:155]: 8f1f011fef11d17dac090609978f0ee2df2b1ceb

[the remark this answers is not in the document]
Comment (Hanooona) [reply internal/bootstrap/data/setting.go:155]: 8f1f011fef11d17dac090609978f0ee2df2b1ceb

[the remark this answers is not in the document]
Comment (Hanooona) [reply internal/bootstrap/data/setting.go:155]: 8f1f011fef11d17dac090609978f0ee2df2b1ceb

[the remark this answers is not in the document]
Comment (Hanooona) [reply internal/bootstrap/data/setting.go:155]: 8f1f011fef11d17dac090609978f0ee2df2b1ceb

[the remark this answers is not in the document]
Comment (Hanooona) [reply internal/bootstrap/data/setting.go:155]: 8f1f011fef11d17dac090609978f0ee2df2b1ceb

[the remark this answers is not in the document]
Comment (Hanooona) [reply internal/bootstrap/data/setting.go:155]: 8f1f011fef11d17dac090609978f0ee2df2b1ceb

[the remark this answers is not in the document]
Comment (Hanooona) [reply internal/bootstrap/data/setting.go:155]: 8f1f011fef11d17dac090609978f0ee2df2b1ceb

[the remark this answers is not in the document]
Comment (Hanooona) [reply internal/bootstrap/data/setting.go:155]: 8f1f011fef11d17dac090609978f0ee2df2b1ceb

[the remark this answers is not in the document]
Comment (Hanooona) [reply internal/bootstrap/data/setting.go:155]: 8f1f011fef11d17dac090609978f0ee2df2b1ceb

[the remark this answers is not in the document]
Comment (Hanooona) [reply internal/bootstrap/data/setting.go:155]: 8f1f011fef11d17dac090609978f0ee2df2b1ceb

[the remark this answers is not in the document]
Comment (Hanooona) [reply internal/bootstrap/data/setting.go:155]: 8f1f011fef11d17dac090609978f0ee2df2b1ceb

[the remark this answers is not in the document]
Comment (Hanooona) [reply internal/bootstrap/data/setting.go:155]: 8f1f011fef11d17dac090609978f0ee2df2b1ceb

[the remark this answers is not in the document]
Comment (Hanooona) [reply internal/bootstrap/data/setting.go:155]: 8f1f011fef11d17dac090609978f0ee2df2b1ceb

[the remark this answers is not in the document]
Comment (Hanooona) [reply internal/bootstrap/data/setting.go:155]: 8f1f011fef11d17dac090609978f0ee2df2b1ceb

[the remark this answers is not in the document]
Comment (Hanooona) [reply internal/bootstrap/data/setting.go:155]: 8f1f011fef11d17dac090609978f0ee2df2b1ceb

[the remark this answers is not in the document]
Comment (Hanooona) [reply internal/bootstrap/data/setting.go:155]: 8f1f011fef11d17dac090609978f0ee2df2b1ceb

[the remark this answers is not in the document]
Comment (Hanooona) [reply internal/bootstrap/data/setting.go:155]: 8f1f011fef11d17dac090609978f0ee2df2b1ceb

[the remark this answers is not in the document]
Comment (Hanooona) [reply internal/bootstrap/data/setting.go:155]: 8f1f011fef11d17dac090609978f0ee2df2b1ceb

[the remark this answers is not in the document]
Comment (Hanooona) [reply internal/bootstrap/data/setting.go:155]: 8f1f011fef11d17dac090609978f0ee2df2b1ceb

[the remark this answers is not in the document]
Comment (Hanooona) [reply internal/bootstrap/data/setting.go:155]: 8f1f011fef11d17dac090609978f0ee2df2b1ceb

[the remark this answers is not in the document]
Comment (Hanooona) [reply internal/bootstrap/data/setting.go:155]: 8f1f011fef11d17dac090609978f0ee2df2b1ceb

[the remark this answers is not in the document]
Comment (Hanooona) [reply internal/bootstrap/data/setting.go:155]: 8f1f011fef11d17dac090609978f0ee2df2b1ceb

[the remark this answers is not in the document]
Comment (Hanooona) [reply internal/bootstrap/data/setting.go:155]: 8f1f011fef11d17dac090609978f0ee2df2b1ceb

[the remark this answers is not in the document]
Comment (Hanooona) [reply internal/bootstrap/data/setting.go:155]: 8f1f011fef11d17dac090609978f0ee2df2b1ceb

[the remark this answers is not in the document]
Comment (Hanooona) [reply internal/bootstrap/data/setting.go:155]: 8f1f011fef11d17dac090609978f0ee2df2b1ceb

[the remark this answers is not in the document]
Comment (Hanooona) [reply internal/bootstrap/data/setting.go:155]: 8f1f011fef11d17dac090609978f0ee2df2b1ceb

[the remark this answers is not in the document]
Comment (Hanooona) [reply internal/bootstrap/data/setting.go:155]: 8f1f011fef11d17dac090609978f0ee2df2b1ceb

[the remark this answers is not in the document]
Comment (Hanooona) [reply internal/bootstrap/data/setting.go:155]: 8f1f011fef11d17dac090609978f0ee2df2b1ceb

[the remark this answers is not in the document]
Comment (Hanooona) [reply internal/bootstrap/data/setting.go:155]: 8f1f011fef11d17dac090609978f0ee2df2b1ceb

[the remark this answers is not in the document]
Comment (Hanooona) [reply internal/bootstrap/data/setting.go:155]: 8f1f011fef11d17dac090609978f0ee2df2b1ceb

[the remark this answers is not in the document]
Comment (Hanooona) [reply internal/bootstrap/data/setting.go:155]: 8f1f011fef11d17dac090609978f0ee2df2b1ceb

[the remark this answers is not in the document]
Comment (Hanooona) [reply internal/bootstrap/data/setting.go:155]: 8f1f011fef11d17dac090609978f0ee2df2b1ceb

[the remark this answers is not in the document]
Comment (Hanooona) [reply internal/bootstrap/data/setting.go:155]: 8f1f011fef11d17dac090609978f0ee2df2b1ceb

[the remark this answers is not in the document]
Comment (Hanooona) [reply internal/bootstrap/data/setting.go:155]: 8f1f011fef11d17dac090609978f0ee2df2b1ceb

[the remark this answers is not in the document]
Comment (Hanooona) [reply internal/bootstrap/data/setting.go:155]: 8f1f011fef11d17dac090609978f0ee2df2b1ceb

[the remark this answers is not in the document]
Comment (Hanooona) [reply internal/bootstrap/data/setting.go:155]: 8f1f011fef11d17dac090609978f0ee2df2b1ceb

[the remark this answers is not in the document]
Comment (Hanooona) [reply internal/bootstrap/data/setting.go:155]: 8f1f011fef11d17dac090609978f0ee2df2b1ceb

[the remark this answers is not in the document]
Comment (Hanooona) [reply internal/bootstrap/data/setting.go:155]: 8f1f011fef11d17dac090609978f0ee2df2b1ceb

[the remark this answers is not in the document]
Comment (Hanooona) [reply internal/bootstrap/data/setting.go:155]: 8f1f011fef11d17dac090609978f0ee2df2b1ceb

[the remark this answers is not in the document]
Comment (Hanooona) [reply internal/bootstrap/data/setting.go:155]: 8f1f011fef11d17dac090609978f0ee2df2b1ceb

[the remark this answers is not in the document]
Comment (Hanooona) [reply internal/bootstrap/data/setting.go:155]: 8f1f011fef11d17dac090609978f0ee2df2b1ceb

[the remark this answers is not in the document]
Comment (Hanooona) [reply internal/bootstrap/data/setting.go:155]: 8f1f011fef11d17dac090609978f0ee2df2b1ceb

[the remark this answers is not in the document]
Comment (Hanooona) [reply internal/bootstrap/data/setting.go:155]: 8f1f011fef11d17dac090609978f0ee2df2b1ceb

**Pull request merged**